[Feature] 이슈 발행 기능을 구현한다.

asd42270 opened the pull request:
# Pull Request 요약

- 이슈 발행 기능을 구현합니다.

## 변경 사항

- 커밋 로그가 이상해 보이지만, 어제 커밋 올렸었는데, 중간에 pr 올라온 거 머지하느라 develop에 옮겼다가 다시 작업해서 그렇습니당...
- ArticleIssuePublishRequestDto: 이슈 발행을 위한 Dto를 생성한다.
- ArticleIssueService: 서비스 클래스를 구현한다. 유지보수와유지관리 이슈 발행을 구분한다.
- ArticleIssueController: 유지보수와 유지관리를 나누어 컨트롤러를 작성한다.
- ArticleIssue: member 컬럼을 추가한다. 유지관리/유지보수 생성을 위한 생성자를 추가한다.
- "기사를 찾을 수 없다"라는 예외의 경우 저도 공통으로 묶어놨었는데 혜미님께서 똑같이 작업해 주셔서...ㅎㅎ 감사합니당
- 아직 게시글 생성에 대한 로직이 없어 포스트맨 테스트는 다 하지 못했습니다. 추후 리팩토링과정 거치면서 작업하겠습니다.
## 관련 이슈

- #92 

## 개발 유형

- [ ] 버그 수정
- [x] 새로운 기능 개발
- [ ] 리팩토링
- [ ] 테스트 코드 작성
- [ ] 문서 업데이트
- 기타 (아래에 설명 추가)

## 작업 기간

- 작업 시작일: (2024-05-22)
- 작업 종료일: (2024-05-23)

## 유의사항

- 이 PR로부터 발생할 수 있는 유의사항이 있나요? 이 PR이 다른 부분에 미칠 수 있는 영향 등을 설명하세요.

## 참고문헌

- 이 PR을 작성하며 참조한 문헌이나 가이드가 있다면 여기에 링크를 추가하세요.

asd42270 committed:
[Feature] 이슈 발행 기능을 구현한다.
- RepairArticleRepository: 유지보수 게시글 조회를 위해 repository를 생성한다.
- ManageArticleRepository: 유지관리 게시글 조회를 위해 repository를 생성한다.
- ArticleIssueRepository: 이슈 repository를 생성한다.
-

asd42270 committed:
[Feature] 이슈 발행 기능을 구현한다.
- ArticleIssue: BaseTimeEntity를 상속한다. BaseTimeEntity의 createAt을 사용하고, 유지관리와 유지보수 게시글 생성을 구분하여 생성자 메서드를 추가한다.
import문을 축약한다.
- ArticleIssueCheck: import문 정리

asd42270 committed:
[Feature] 이슈 발행 기능을 구현한다.
- ArticleIssuePublishRequestDto: 이슈 발행을 위한 Dto를 생성한다.
- ArticleIssueService: 서비스 클래스를 구현한다. 유지보수와유지관리 이슈 발행을 구분한다.
- ErrorCode: ARTICLE_NOT_FOUND를 추가한다.
- ArticleNotFoundException: 공통 클래스로 묶어 사용한다.
- ArticleIssueController: 유지보수와 유지관리를 나누어 컨트롤러를 작성한다.

asd42270 committed:
Merge branch '10aeat:develop' into feature/92

asd42270 committed:
Merge branch 'develop' into feature/92

asd42270 committed:
Merge pull request #1 from asd42270/feature/92

Feature/92

asd42270 committed:
[Feature] 이슈 발행 기능을 구현한다.
- ArticleIssuePublishRequestDto: 이슈 발행을 위한 Dto를 생성한다.
- ArticleIssueService: 서비스 클래스를 구현한다. 유지보수와유지관리 이슈 발행을 구분한다.
- ArticleIssueController: 유지보수와 유지관리를 나누어 컨트롤러를 작성한다.
- ArticleIssue: member 컬럼을 추가한다. 유지관리/유지보수 생성을 위한 생성자를 추가한다.

Galmaeki commented on src/main/java/com/final_10aeat/domain/articleIssue/dto/ArticleIssuePublishRequestDto.java:
```diff
@@ -0,0 +1,7 @@
+package com.final_10aeat.domain.articleIssue.dto;
+
+public record ArticleIssuePublishRequestDto(
+        String title,
```
Validation이 있으면 좋을것같습니다!

Galmaeki reviewed (approved):
고생 많으셨습니다!
코멘트 확인 부탁드려요!

asd42270 committed:
[Feature] 이슈 발행 기능을 구현한다.
- ArticleIssuePublishRequestDto: NotBlank 어노테이션 추가
- ArticleIssueController: Valid 어노테이션 추가

ghrltjdtprbs commented on src/main/java/com/final_10aeat/global/exception/ErrorCode.java:
```diff
@@ -37,6 +35,9 @@ public enum ErrorCode {
     MEMBER_MISMATCH(HttpStatus.CONFLICT, "일치하지 않는 사용자입니다."),
     DISAGREE_TERM(HttpStatus.CONFLICT, "약관에 동의해야 합니다."),
 
+    // ManagerArticle/RepairArticle
```
게시글 공통 메시지이므로 // ARTICLES로 통일하는게 더 보기 좋을 것 같습니다!

ghrltjdtprbs reviewed (approved):
수고하셨습니다! 
커밋 내역을 봤는데, 현재 커밋 컨벤션이 지켜지지 않고 있는 것 같습니다! 기능 추가는 [feat] 입니다!
디스커션에 코드 컨벤션 올려져 있으니 참고해주세요 ㅎㅎ 😊😊 
<img width="436" alt="스크린샷 2024-05-24 10 44 59" src="https://github.com/10aeat/10aeat_Backend/assets/105612931/2d10a7a1-cc26-43a3-af0c-c1dea4585761">

asd42270 committed:
[fix] 이슈 발행 기능을 구현한다.
- ErrorCode: 에러코드 주석 변경
